### DISCUSSION TITLE
Apply custom cumulative factor to the latest diagonal (Chain Ladder Example from Wikipedia)

### DISCUSSION THREAD

**Discussion by alexbcz** [casact/chainladder-python#479] [Q&A]
Hi Everyone,

I'm trying to reproduce the example showed on the wikipedia page about Chain Ladder : https://en.wikipedia.org/wiki/Chain-ladder_method#Example

I could reproduce all the calculations showed on the page, but I'm still missing the last table : Estimation of ultimate claims

I know how to get the cdf I want to apply:
`cl.Development(average="volume", n_periods=3).fit(xyz_tri["Incurred"]).cdf_`

I know how to get the latest diagnoal: 
`xyz_tri.latest_diagonal`

But how to apply the cdf to the latest diagonal in order to get the latest table of the example?
I think, I'm also missing how to calculate the tail factor?
Another question: Can I apply the cdf to the whole triangle to get the full triangle based on these cdf?

Here's what I have done so far : https://github.com/alexbcz/wikipedia_chain_ladder_example/blob/main/Wikipedia%20ChainLadder%20example.ipynb

Thank in advance

**Comment by jbogaardt** [comment]
Sounds like you want the multiplicative `Chainladder` estimator.  I would refer you to @kennethshsu's excellent [tutorial](https://chainladder-python.readthedocs.io/en/latest/getting_started/tutorials/deterministic-tutorial.html) on the subject.

**Reply by alexbcz** [reply]
Thanks a lot John,
It worked perfectly.
If there are folks willing to do the same, I updated the notebook with the solution provided : https://github.com/alexbcz/wikipedia_chain_ladder_example/blob/main/Wikipedia%20ChainLadder%20example.ipynb

I have an other question, once I did the job with the reported claims, I wanted to do the same with the paid claims.
The full example is not on Wikipedia, I found it on the casact website : https://www.casact.org/sites/default/files/database/studynotes_friedland_estimating.pdf
Whereas with the reported claims no tail factor was applied, for the paid claims they applied a tail factor of 1.002
As no specific tailing method is mentionned in the document, I used `cl.TailConstant(1.002)`
I got almost the same results than the document for the utimate paid claims (less than 0.05% of difference in the worst case).
But I am bit confused, because when I look at the ldfs of my model, I can't see my custom tail factor :

|       | 12-24 | 24-36 | 36-48 | 48-60 | 60-72 | 72-84 | 84-96 | 96-108 | 108-120 | 120-132 | 132-144 |
|------:|------:|------:|------:|------:|------:|------:|------:|-------:|--------:|--------:|--------:|
| (All) | 1.702 | 1.186 | 1.091 | 1.044 | 1.019 | 1.009 | 1.005 |  1.002 |   1.002 |   1.001 |   1.001 |

Am I missing something?

**Reply by hoverat** [reply]
Did you look at your cdfs?

**Reply by alexbcz** [reply]
Here is what I got for the cdfs:
	12-Ult	24-Ult	36-Ult	48-Ult	60-Ult	72-Ult	84-Ult	96-Ult	108-Ult	120-Ult	132-Ult
(All)	2.391	1.404	1.184	1.085	1.040	1.021	1.011	1.006	1.004	1.002	1.001

Here we can see the tail factor of 1.002
So everything looks good 👍 
Thank you and sorry for the confusion 😅
